File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1120_문자열/1120.cpp
#include <algorithm>
#include <cstring>
#include <stdio.h>

using namespace std;

char A[52], B[52];

int get_ans(int len, int idx)
{
    int cnt = 0;

    for (int j = 0; j < len; ++j)
        if (A[j] != B[j + idx])
            ++cnt;
    return cnt;
}

int main()
{
    int diff, res = 50;

    scanf("%s %s", A, B);
    diff = strlen(B) - strlen(A);
    for (int i = 0; i <= diff; ++i)
        res = min(res, get_ans(strlen(A), i));
    printf("%d\n", res);
    return 0;
}

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1120_문자열/1120.py
A, B = input().split()
res = 50

for i in range(len(B) - len(A) + 1):
    cnt = 0
    for j in range(len(A)):
        if A[j] != B[i + j]:
            cnt += 1
    res = min(res, cnt)
print(res)
